NetworkLayer.listen: don't "die" when node backend fails

rvl opened the pull request:
Part of investigations into #100.

# Overview

"die" will exit the process, which might not be what we want in all cases. Throw an exception instead, which will also result in the process exiting, but could be caught if needed.

KtorZ commented on test/integration/Main.hs:
```diff
@@ -2,33 +2,46 @@ module Main where
         ]
-    waitForNode (_client ctx) (MaxWaitingTime 90)
+    let wClient = mkHttpClient (toBaseUrl $ env ! "WALLET_ADDRESS") manager
+    let dClient = mkHttpDocClient (toBaseUrl $ env ! "WALLET_DOC_ADDRESS") manager
```
NOTE: We are defaulting these two values, so are sure they're there!

KtorZ commented on test/integration/Test/Integration/Documentation.hs:
```diff
@@ -0,0 +1,17 @@
+spec client = do
+    it "Fetches the documentation from the API" $ do
+        response <- runExceptT $ Client.getSwaggerJson client
+        response `shouldSatisfy` isRight
```
By fetching the documentation, we make sure that it actually gets rendered properly (basically invoking a whole bunch of `Example` classes as well as pretty much everything in `Swagger.hs`). We have quite a few "unsafe" things there that aren't caught at runtime. However, they'll also prevent the doc from being rendered, so if this test passes, we can assume that things are fine.

KtorZ commented on test/integration/Test/Integration/Framework/Cluster.hs:
```diff
@@ -85,17 +84,15 @@ startCluster nodes = do
-                yield (Just client) >> startWallet node nodeEnv
-    let client = Prelude.head $ catMaybes (map snd handles)
+                yield (Just (nodeEnv, manager)) >> startWallet node nodeEnv
+    let (env, manager) = Prelude.head $ catMaybes (map snd handles)
```
So basically, we are passing around the `Manager` and defer the creation of the client. We'll use the same manager to instantiate both a wallet client and a doc client (since the underlying server uses the same TLS certificates anyway!)

KtorZ commented on test/integration/Test/Integration/Framework/DSL.hs:
```diff
@@ -158,7 +158,7 @@ data Setup = Setup
         :: AssuranceLevel
     , _mnemonicWords
-        :: Maybe [Text]
+        :: [Text]
```
The reason for this is purely esthetic. `mnemonicWords .~ ["patate", "autruche", ...]` reads better than `mnemonicWords .~ Just ["patate", "autruche" ....`. And I don't want us to be introducing every possible operator from the lens library (So yes, I am aware of `(?~)`). But let's leave it simple :)

KtorZ commented on test/integration/Test/Integration/Framework/DSL.hs:
```diff
@@ -393,14 +392,13 @@ expectAddressInIndexOf = \case
     -> Either ClientError Transaction
     -> m ()
-expectTxStatusEventually w statuses = \case
+expectTxStatusEventually statuses = \case
```
So the `Wallet` here was only needed because of the bug described in #141. Now that is is fixed, the parameter is superfluous.

rvl committed:
NetworkLayer.listen: don't "die" when node backend fails

"die" will exit the process, which might not be what we want in all
cases. Throw an exception instead, which will also result in the
process exiting, but could be caught if needed.
